substrate/primitives/metadata-ir/src/types.rs

```diff
@@ -118,6 +120,87 @@ impl IntoPortable for RuntimeApiMethodParamMetadataIR {
 	}
 }
 
+/// Metadata of the the runtime query dispatch.
+#[derive(Clone, PartialEq, Eq, Encode, Debug)]
+pub struct RuntimeQueryIR<T: Form = MetaForm> {
+	/// The type implementing the runtime query dispatch.
+	pub ty: T::Type,
+	/// The query interfaces metadata.
+	pub interfaces: Vec<QueryInterfaceIR<T>>,
+}
+
+/// Metadata of a runtime query interface.
+#[derive(Clone, PartialEq, Eq, Encode, Debug)]
+pub struct QueryInterfaceIR<T: Form = MetaForm> {
```

Comment:
How much could we align this with Runtime APIs in terms of the naming of structts/fields?

For runtime APIs we have:

```
RuntimeApiMetadataIR { name, methods, docs, deprecation_info }
RuntimeApiMethodMetadataIR { name, inputs, output, docs, deprecation_info }
RuntimeApiMethodParamMetadataIR { name, ty }
```

So here, we could, per pallet, have something like:

```
struct PalletApiMethodMetadataIR { name, id?, inputs: Vec<PalletApiMethodParamMetadataIR>, output, docs }
struct PalletApiMethodParamMetadataIR { name, ty }
```

(I went for `Api` instead of `ViewFunction` just because I prefer it to viewFunctions and it's shorter, but it's all good!)

I guess having the `id` field lets us avoid having to construct it, since it's a bit arduous hashing the relevant bits, so makes sense for now! Personally I'd simplify the way we call these things to be more like runtime APIs, but I'm happy to propose that as a separate PR!

Reply:
I agree should try to align the metadata with Runtime APIs...in theory we should be able to replace Runtime APIs in the future with "view functions".

As for naming...it would be good to come up with something consistent: as you can see I am using "query" interchangeably with "view_function". API is okay but maybe too generic and could be confused with existing Runtime API? Naming is hard, any suggestions welcome.

Reply:
I personally prefer Api to align with Runtime APIs, but happy for everything to be named consistently with any of `Api`, `Query`, `ViewFunc` (just to help shorten a touch) or `ViewFunction`. For the sake of being able to rename it all quite easily in a followup PR if we like, perhaps one of the latter two is ideal (easy to search and replace `ViewFunc`/`ViewFunction`!)